hack/drone.jsonnet

```diff
@@ -0,0 +1,323 @@
+//     DOCKER_PASSWORD:
+//       from_secret: docker_password
+//   commands:
+//     - make gitmeta
+//     - make login
+//     - make push
+//   volumes:
+//     - name: dockersock
+//       path: /var/run
+//   when:
+//     event:
+//       - push
+//   depends_on:
+//     - test.Integration
+// };
+
+local notify = {
+  name: "slack",
+  image: "plugins/slack",
+  settings:
+    {
+     webhook: {from_secret: "slack_webhook"},
+     channel: "proj-talos-maint",
+    },
+};
+
+// Functions
+
+local step(name, depends_on=[], environment={}, target="") = {
+  name: std.format("%s", name),
+  image: build_container,
+  commands: [if target == "" then std.format("make %s", name) else std.format("make %s", target)],
+  environment: build_env_vars+environment,
+  volumes: step_volumes,
+  depends_on: [x.name for x in depends_on],
+};
+
+
+local pipeline(name, steps=[]) = {
+  kind: "pipeline",
+  name: name,
+  node: node,
+  services: if name != "notify" then [docker] else [],
+  volumes: if name != "notify" then pipeline_volumes else [],
+  clone: {
+    disable: true,
+  },
+  steps: if name != "notify" then [clone, buildkit] + steps else steps,
+};
+
+// Apps
+
+local machined = step("machined", [clone]);
+local osd = step("osd", [clone]);
+local trustd = step("trustd", [clone]);
+local proxyd = step("proxyd", [clone]);
+local ntpd = step("ntpd", [clone]);
+local osctl_linux = step("osctl-linux", [clone]);
+local osctl_darwin = step("osctl-darwin", [clone]);
+
+// Artifacts
+
+local rootfs =  step("rootfs", [machined, osd, trustd, proxyd, ntpd]);
+local initramfs = step("initramfs", [rootfs]);
+local installer = step("installer", [rootfs]);
+local container = step("container", [rootfs]);
+local image_azure = step("image-azure", [installer]);
+local image_gce = step("image-gce", [installer]);
+local kernel = step("kernel", [clone]);
+local iso = step("iso", [installer]);
+
+
+// CAPI.
+
+local push_image_azure = step("push-image-azure", [image_azure], creds_env_vars);
+local push_image_gce = step("push-image-gce", [image_gce], creds_env_vars);
+
+// Tests
+
+local lint = step("lint", [rootfs]);
+local unit_tests = step("unit-tests", [rootfs]);
+local basic_integration = step("basic-integration", [container, osctl_linux]);
+local capi = step("capi", [basic_integration], creds_env_vars);
+local e2e_integration_azure = step("e2e-integration-azure", [capi, push_image_azure], {PLATFORM: "azure"}, "e2e-integration");
+local e2e_integration_gce = step("e2e-integration-gce", [capi, push_image_gce], {PLATFORM: "gce"}, "e2e-integration");
+local conformance_azure = step("conformance-azure", [capi, push_image_azure], {PLATFORM: "azure", CONFORMANCE: "run"}, "e2e-integration");
+local conformance_gce = step("conformance-gce", [capi, push_image_gce], {PLATFORM: "gce", CONFORMANCE: "run"}, "e2e-integration");
+
+// Steps
+
+local apps = [
+    machined,
+    osd,
+    trustd,
+    proxyd,
+    ntpd,
+    osctl_linux,
+    osctl_darwin,
```

Comment:
nit, but having osctl_darwin here causes it to get built during e2e tests which isn't needed. not super significant and probably causes more complexity to filter it out

Reply:
Yeah, it really makes the implementation uglier if we try to pull that bit out. I tried to design in in such a way that each pipeline sort of builds on the previous one. Plus this gets us closer to running a more complete e2e test.